### PR TITLE
[AAE-22900] Update ObjectDataTableAdapter to support server and clien…

### PR DESCRIPTION
**Pull request by amolodyh-hyland**
Fixes [AAE-22900](https://hyland.atlassian.net/browse/AAE-22900?atlOrigin=eyJpIjoiYmU2M2FhMzFmMjFmNGVjNDkxOGRmMmNhOGI2MGE5Y2YiLCJwIjoiaiJ9). This adds server sorting for the base `ObjectDataTableAdapter` to enable tables in tabs such as "Process Admin" User Tasks, Services Tasks, and Applications List.

**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [x] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [x] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
 [AAE-22900](https://hyland.atlassian.net/browse/AAE-22900?atlOrigin=eyJpIjoiYmU2M2FhMzFmMjFmNGVjNDkxOGRmMmNhOGI2MGE5Y2YiLCJwIjoiaiJ9)

**What is the new behaviour?**

Allows for enabling server sorting where it the adapter does not sort the data. The server data is rendered in the order it comes in.

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

This a partial fix to the provided story  [AAE-22900](https://hyland.atlassian.net/browse/AAE-22900?atlOrigin=eyJpIjoiYmU2M2FhMzFmMjFmNGVjNDkxOGRmMmNhOGI2MGE5Y2YiLCJwIjoiaiJ9). After this is merged, hxp-frontend-apps will need to be updated by updating the `applications-list-filtered-data-adapter` adapter as well.

[AAE-22900]: https://hyland.atlassian.net/browse/AAE-22900?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[AAE-22900]: https://hyland.atlassian.net/browse/AAE-22900?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: lib/process-services-cloud/src/lib/process/process-list/datatable/process-list-datatable-adapter.ts [hunk 1]
```diff
@@ -21,7 +21,7 @@ import { ProcessInstanceCloudListViewModel } from '../models/perocess-instance-c
 
 export class ProcessListDatatableAdapter extends ObjectDataTableAdapter {
     constructor(data: ProcessInstanceCloudListViewModel[], schema: DataColumn<ProcessListDataColumnCustomData>[]) {
```
Comment (amolodyh-hyland) [inline lib/process-services-cloud/src/lib/process/process-list/datatable/process-list-datatable-adapter.ts:23]: @DenysVuika @eromano Do we want to provide the ability to set the `sortingMode` in these extending classes so we can utilize feature flags in the hxp-frontend-apps projects? Not sure if we want a feature flag for this but if we did, we'd need to be able to provide the sort option here.

Based on the information I've received, we don't want pages using these adapters to use client sorting so the constructor parameter would be redundant. Also, we would have to remove the constructor parameter once the feature flag is removed so it seems architecturally redundant.

File: lib/core/src/lib/datatable/data/object-datatable-adapter.ts [hunk 2]
```diff
@@ -54,9 +57,10 @@ export class ObjectDataTableAdapter implements DataTableAdapter {
         return schema;
     }
 
-    constructor(data: any[] = [], schema: DataColumn[] = []) {
+    constructor(data: any[] = [], schema: DataColumn[] = [], sortingMode: SortingMode = 'client') {
         this._rows = [];
         this._columns = [];
+        this._sortingMode = sortingMode?.toString().toLowerCase() === 'server' ? 'server' : 'client';
```
Comment (sdonnell-hyland) [inline lib/core/src/lib/datatable/data/object-datatable-adapter.ts:63]: nit: toString() probably isn't necessary

Comment (amolodyh-hyland) [reply lib/core/src/lib/datatable/data/object-datatable-adapter.ts:63]: Just trying to be super careful incase `sortingMode` is not provided as a string. You think that's overkill?

Comment (sdonnell-hyland) [reply lib/core/src/lib/datatable/data/object-datatable-adapter.ts:63]: It doesn't really hurt anything, but I would expect all consumers of ObjectDataTableAdapter to be using TypeScript typings, and not need the additional javascript safety checks.  I spot checked some files and didn't see anywhere else where typings weren't trusted, but I could have easily missed another file that does.

Personally, I would just account for sortingMode being undefined, but if it is defined assume its a SortingMode.

**Review by sdonnell-hyland — APPROVED** [review]
Looks good to me

**Review by jcain82 — APPROVED** [review]

File: lib/core/src/lib/datatable/data/object-datatable-adapter.ts (hunk 2 above)
Comment (amolodyh-hyland) [reply lib/core/src/lib/datatable/data/object-datatable-adapter.ts:63]: Sounds good, makes sense to me. I'll update it.

**Pull request closed without merging**